internal/slicer/slicer.go

```diff
@@ -291,16 +321,47 @@ func Run(options *RunOptions) (*Report, error) {
 		}
 		err := scripts.Run(&opts)
 		if err != nil {
-			return nil, fmt.Errorf("slice %s: %w", slice, err)
+			return fmt.Errorf("slice %s: %w", slice, err)
 		}
 	}
 
 	err = removeAfterMutate(targetDir, knownPaths)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	return report, nil
+	manifestSlices := manifest.LocateManifestSlices(options.Selection.Slices, manifestFilename)
+	if len(manifestSlices) > 0 {
+		var files []io.Writer
+		for relPath := range manifestSlices {
+			logf("Generating manifest at %s...", relPath)
+			absPath := filepath.Join(options.TargetDir, relPath)
+			if err := os.MkdirAll(filepath.Dir(absPath), 0755); err != nil {
+				return err
+			}
+			file, err := os.OpenFile(absPath, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, manifestMode)
+			if err != nil {
+				return err
+			}
+			files = append(files, file)
+			defer file.Close()
+		}
+		w, err := zstd.NewWriter(io.MultiWriter(files...))
+		if err != nil {
+			return err
+		}
+		defer w.Close()
+		writeOptions := &manifest.WriteOptions{
+			ManifestSlices: manifestSlices,
+			PackageInfo:    pkgInfos,
+			Selection:      options.Selection.Slices,
+			Report:         report,
+			ManifestMode:   manifestMode,
+		}
+		err = manifest.Write(writeOptions, w)
```

Comment:
This whole logic above, including the new use of report and fsutil, should probably be on its own function instead of making this large function even larger.  Note that we don't need the *Options sort of type for this kind of internal organization, since this is all entirely private and there'll be a single call to that function anyway.

Reply:
Done in https://github.com/canonical/chisel/pull/142/commits/e07033267341739c2862d4388efc9760c2f32b52.